torch/nn/parallel/distributed.py

```diff
@@ -1313,6 +1329,22 @@ def _check_comm_hook(self, hook):
     def _distributed_rank(self):
         return dist.get_rank(self.process_group)
 
+    def _create_gloo_pg(self):
+        """
+        Creates a gloo process group with same ranks as self.process_group to be
+        used for debugability such as monitored_barrier().
+        """
+        if _GLOO_AVAILABLE and dist._get_debug_mode() != dist._DistributedDebugLevel.OFF:
```

Comment:
feel like this condition can be removed so that _create_gloo_pg() can be reused for other cases